warehouse/forklift/legacy.py

```diff
@@ -1252,8 +1172,9 @@ def file_upload(request):
                 k: h.hexdigest().lower() for k, h in metadata_file_hashes.items()
             }
 
+        attestations: list[Attestation] | None = None
```

Comment:
Nitpick: No need for a double state here, I think (both empty list and `None`):

```suggestion
        attestations: list[Attestation] = []
```